### ISSUE TITLE
FP-30: piano.write_register(RolandAddressMap.metronomeStatus,1) not working

### ISSUE THREAD

**Issue by greczimarton** [evanraalte/RolandPiano#1]
Hi,

Thanks for the amazing repo (both this and RolandPianoControl).

The code works perfectly when it changes the volume of the piano (read and write). It also works well when reading the data inside the registers.
But when I would like to turn on the metronome with code, it doesn't work.
Only the write method doesn't work, read method returns the correct value when I turn on and off the metronome manually on the piano.

Can I ask you for any ideas on this?

Also where did you find the address map for the piano? I checked the roland midi implementation for both fp-10 and fp-30 and I haven't found anything on metronome for example. May I ask for the resources that you use for this project? Mainly on the midi implementation of roland?

I would like to create a similar project: ESP32-S microcontroller based webapp, that controls the piano through bluetooth.

Also if there are any way I could contribute to this project please let me know. :)

Thanks,
Marton

**Comment by evanraalte** [comment]
Hi Marton, thanks for your interest in this project! 

For some reason I did not see that someone left a comment. I will try to set-up notifications correctly :). It has been a while ago since I implemented the API's, but I think the metronome was working on my end. I will do some testing and will hopefully report back with a working example.

The address map was obtained by unpacking the Android application that is provided by Roland. From here I also deduced a part of the command sequences. 

There are still some mysteries to resolve, like pushing setting changes from the piano itself (right now I need to poll), so feel free to contribute to this project!
